errdefs/invalid.go

```diff
@@ -0,0 +1,69 @@
+package errdefs
+
+import (
+	"errors"
+	"fmt"
+)
+
+// InvalidInput is an error interface which denotes whether the opration failed due
+// to a the resource not being found.
+type ErrInvalidInput interface {
+	InvalidInput() bool
+	error
+}
+
+type invalidInputError struct {
+	cause error
```

Comment:
Any reason to not just leave this as an embedding, so Error() doesn't have to be reimplemented?

Reply:
I want to be able to implement the causal interface, embedding `error` I don't think we'd be able to get at the underlying error type.

Reply:
```
package main

import (
	"fmt"
	"errors"
)

func main() {
	var e error = &invalidInputError{errors.New("Sargun's error")}
	fmt.Println(e)
	causalError, ok := e.(Causal)
	fmt.Println(ok)
	fmt.Println(causalError.Cause())
}

type Causal interface {
	Cause() error
	error
}


type ErrInvalidInput interface {
	InvalidInput() bool
	error
}

type invalidInputError struct {
	error
}

func (e *invalidInputError) Cause() error {
	return e.error
}
```

Implements ^ the error interface, and don't have to restate Error. Also, can fetch the underlying error.

Reply:
🐑 
Wow yes, not sure what I was thinking.